### PR TITLE
Add https

### LINKED ISSUE #33: Setup HTTPS
Currently the website is using HTTP because it was easier to setup at the beginning.
But for obvious security reason, HTTPS must be setup to encrypt the traffic.

### PR DESCRIPTION
**Pull request by atxr**
This PR adds HTTPS certificates generation to the deployment pipeline for production mode.
I used `certbot` to generate certificates inside a docker container, with shared volumes with the `nginx` container.
These shared volumes allow `nginx` to load certificates generated by `certbot` inside the configuration and setup ssl.
SSL cipher suits and protocols used are generated by certbot/letsencrypt and are supposed to be the latest one.
This PR will close #33

### CODE REVIEW

**Comment by atxr** [comment]
Ready for reviews! :rocket:

File: nginx/Dockerfile [hunk 1]
```diff
@@ -6,7 +6,10 @@ ENV NODE_PORT=$NODE_PORT
 ARG REACT_PORT
 ENV REACT_PORT=$REACT_PORT
 
+ARG SERVER_NAME
+ENV SERVER_NAME=$SERVER_NAME
```
Comment (amtoine) [inline nginx/Dockerfile:10]: the expansion works fine here? :thinking: 
`$SERVER_NAME` and not `${SERVER_NAME}`? :yum:

File: docker-compose.yml [hunk 2]
```diff
@@ -27,5 +27,17 @@ services:
       args:
         - NODE_PORT=$NODE_PORT
         - REACT_PORT=$REACT_PORT
+        - SERVER_NAME=$SERVER_NAME
     ports:
       - 80:80
+      - 443:443
+    volumes:
+      - ./certbot/www:/var/www/certbot/:ro
+      - ./certbot/conf/:/etc/nginx/ssl/:ro
```
Comment (amtoine) [inline docker-compose.yml:36]: so here, `nginx` can "read-only" and `certbot` can "read/write"?

File: README.md [hunk 3]
```diff
@@ -107,12 +107,44 @@ Before deploying the application, you need to set the environment variables, as
 cp .env.example .env.production
 ```
 
+#### SSL certification
+
+To setup HTTPS, you will need valid SSL certificates. If you deploy the app for the first time, follow these instructions:
+- Comment or delete the whole server section about 443 in the `nginx/nginx.conf.template` file.
+```
+- server {
```
Comment (amtoine) [inline README.md:115]: what are the `-` at the beginning?
it does not look like a "diff", does it? :open_mouth: 

if it's meant to be a "diff", please write
```
```diff
- server {
- listen 443 default_server ssl http2;
- ...
```
to have the output colored in red :wink:

**Review by amtoine — COMMENTED** [review]
well that looks great :ok_hand: 

i trust you when saying this works on the server side :+1: 

i'll wait for answers to my threads and then approve that :wink:

File: nginx/Dockerfile (hunk 1 above)
Comment (atxr) [reply nginx/Dockerfile:10]: Yesyes no problem no need here!

File: docker-compose.yml (hunk 2 above)
Comment (atxr) [reply docker-compose.yml:36]: Exact! `certbot` will write the certificates in the volume and `nginx` read them

File: README.md (hunk 3 above)
Comment (atxr) [reply README.md:115]: Fix [1cfe0df](https://github.com/iScsc/iscsc.fr/pull/34/commits/1cfe0df6adf500938942d93b374f99ebf266bcc3)

File: nginx/Dockerfile (hunk 1 above)
Comment (amtoine) [reply nginx/Dockerfile:10]: coool :ok_hand: 
better than `bash` then :eyes:

File: docker-compose.yml (hunk 2 above)
Comment (amtoine) [reply docker-compose.yml:36]: got it :+1:

File: README.md (hunk 3 above)
Comment (amtoine) [reply README.md:115]: perfect thanks :yum:

**Review by amtoine — APPROVED** [review]
thanks for the answers and the change :ok_hand: 

that's good for me and my understanding of the `docker` setup :+1:

**Review by ctmbl — APPROVED** [review]
This looks fine to me as far as I can understand it!  
However I have two points here:
 - can you send some `certbot` documentation to allow us to verify that the configuration has been done well?
 - how did you ensure that it works? I guess you juste deployed it and access it from a browser, should we check that the traffic is actually encrypted? should we check that it is done securely/perform a security audit? or does it seems useless to you?
 
 Once again tks for your amazing work!!

**Comment by atxr** [comment]
> can you send some `certbot` documentation to allow us to verify that the configuration has been done well?

Here is [the repo](https://github.com/certbot/certbot)
But you can `tldr certbot`
I mainly used [this article](https://mindsers.blog/post/https-using-nginx-certbot-docker) to setup HTTPS if you want to check it.

>  how did you ensure that it works? I guess you juste deployed it and access it from a browser, should we check that the traffic is actually encrypted? should we check that it is done securely/perform a security audit? or does it seems useless to you?

No actually I just deployed it, and browsed to https://iscsc.fr
I checked that the little locker :lock: was enabled in firefox, which means everything is good with the certificates.
You can actually click on this locker on every https website and add lots of information about the certificates. I put some screenshots for github.com, if you click on `view certificates` you'll have all the technical details.
image

You can check with wireshark how the traffic is encrypted but it's far away from the scope of this PR. However, doing it is very interesting if you are curious :wink: 
*I can also do it during a cybersecurity presentation.*

**Comment by amtoine** [comment]
i'm good with what @atxr said above :ok_hand: 
@ctmbl, feel free to merge when you have all your answers :wink:

**Comment by ctmbl** [comment]
> Here is [the repo](https://github.com/certbot/certbot)
But you can tldr certbot
I mainly used [this article](https://mindsers.blog/post/https-using-nginx-certbot-docker) to setup HTTPS if you want to check it.

Thanks a lot it helped me better understand it :wink: 

>  I can also do it during a cybersecurity presentation.

I would like that!

**Pull request merged**